### PR TITLE
Fixing phpstan issues

### LINKED ISSUE #106: Add static code analyse
Related lib: https://github.com/phpstan/phpstan

### PR DESCRIPTION
**Pull request by mamazu**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Related tickets | fixes #106 
| License         | MIT

Fixing phpstan errors.

### CODE REVIEW

File: src/ApiTestCase.php [hunk 1]
```diff
@@ -108,10 +110,19 @@ public function setUpClient()
...
+            /** @var EntityManager $entityManager */
+            $entityManager = $container->get('doctrine.orm.entity_manager');
+            Assert::notNull($entityManager);
+
+            $this->entityManager = $entityManager;
             $this->entityManager->getConnection()->connect();
 
-            $this->fixtureLoader = static::$sharedKernel->getContainer()->get('fidry_alice_data_fixtures.loader.doctrine');
+            /** @var LoaderInterface $fixtureLoader */
+            $fixtureLoader        = $container->get('fidry_alice_data_fixtures.loader.doctrine');
```
Comment (lchrusciel) [inline src/ApiTestCase.php]:
```suggestion
            $fixtureLoader = $container->get('fidry_alice_data_fixtures.loader.doctrine');
```

File: src/ApiTestCase.php [hunk 2]
```diff
@@ -338,13 +355,17 @@ protected function getFixtureLoader()
     /**
      * @return EntityManager
      */
-    protected function getEntityManager()
+    protected function getEntityManager(): EntityManager
     {
-        if (null === $this->entityManager || !$this->entityManager->getConnection()->isConnected()) {
+        $entityManager = $this->entityManager;
+        if (null === $entityManager || !$entityManager->getConnection()->isConnected()) {
             static::fail('Could not establish test database connection.');
```
Comment (lchrusciel) [inline src/ApiTestCase.php]:
```suggestion
			$this->fail('Could not establish test database connection.');
```

File: src/ApiTestCase.php [hunk 3]
```diff
@@ -406,8 +427,11 @@ private function getMockedResponsesFolder()
      */
     private function getCalledClassFolder()
     {
-        $calledClass = get_called_class();
-        $calledClassFolder = dirname((new \ReflectionClass($calledClass))->getFileName());
+        $calledClass       = get_called_class();
+
+        /** @var string $fileName */
+        $fileName          = (new \ReflectionClass($calledClass))->getFileName();
```
Comment (lchrusciel) [inline src/ApiTestCase.php]: I would prefer to not align these assignments

File: src/ApiTestCase.php [hunk 4]
```diff
@@ -429,6 +453,9 @@ private function assertSourceExists($source)
      */
     private function getRootDir()
     {
-        return $this->get('kernel')->getRootDir();
+        /** @var KernelInterface $kernel
+         */
+        $kernel = $this->get('kernel');
+        return $kernel->getRootDir();
```
Comment (lchrusciel) [inline src/ApiTestCase.php]:
```suggestion

        return $kernel->getRootDir();
```

File: src/JsonApiTestCase.php [hunk 5]
```diff
@@ -37,12 +37,12 @@ protected function buildMatcher()
      * If statusCode is set, asserts that response has given status code.
      *
      * @param Response $response
-     * @param string|null $filename
-     * @param int|null $statusCode
+     * @param string $filename
+     * @param int $statusCode (optional)
```
Comment (lchrusciel) [inline src/JsonApiTestCase.php]:
```suggestion
     * @param int $statusCode
```

And to be honest, I would prefer to remove all redundant doc blocks

**Comment by lchrusciel** [comment]
Thank you, @mamazu! :1st_place_medal:

File: src/JsonApiTestCase.php (hunk 5 above)
Comment (mamazu) [reply src/JsonApiTestCase.php]: If you'd be using ECS in this project there would be a fixer for that. :D

File: src/ApiTestCase.php (hunk 3 above)
Comment (mamazu) [reply src/ApiTestCase.php]: Sorry, Phpstorm was that and the CI didn't catch it.

File: src/JsonApiTestCase.php (hunk 5 above)
Comment (lchrusciel) [reply src/JsonApiTestCase.php]: It's already there! #137 :)

**Pull request merged**